### PR TITLE
Update docker file for windows build

### PR DESCRIPTION
**Pull request by TravisEz13**
reducing the base image size should make the build a little faster
Also, remove the duplicate upload of a file

### CODE REVIEW

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/DockerFile [hunk 1]
```diff
@@ -1,11 +1,30 @@
...
+SHELL ["C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe", "-command"]
+# Install Git, NuGet and cMake
+# Git installs to C:\Program Files\Git
+# nuget installs to C:\ProgramData\chocolatey\bin\NuGet.exe
+# Cmake installs to C:\Program Files\CMake\
+COPY dockerInstall.psm1 containerFiles/dockerInstall.psm1
+RUN Import-Module ./containerFiles/dockerInstall.psm1; `
+    Install-ChocolateyPackage -PackageName git -Executable git.exe; `
+    Install-ChocolateyPackage -PackageName nuget.commandline -Executable nuget.exe; `
+    Install-ChocolateyPackage -PackageName cmake.install -Executable cmake.exe -Cleanup
```
Comment (daxian-dbw) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/DockerFile]: We don't need cmake anymore :)

**Comment by TravisEz13** [comment]
@daxian-dbw @adityapatwardhan  Can you update your reviews?

**Comment by TravisEz13** [comment]
@daxian-dbw @adityapatwardhan  Ping

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/DockerFile [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+FROM microsoft/windowsservercore
 LABEL maintainer='PowerShell Team <powershellteam@hotmail.com>'
 LABEL description="This Dockerfile for Windows Server Core with git installed via chocolatey."
 
-SHELL ["powershell"]
+SHELL ["C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\powershell.exe", "-command"]
+# Install Git, NuGet and cMake
+# Git installs to C:\Program Files\Git
+# nuget installs to C:\ProgramData\chocolatey\bin\NuGet.exe
+# Cmake installs to C:\Program Files\CMake\
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/DockerFile]: Remove cmake comment.

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/dockerInstall.psm1 [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        $exeSource = Get-ChildItem -path "$env:ProgramFiles\$Executable" -Recurse -ErrorAction SilentlyContinue | Select-Object -First 1 -ExpandProperty FullName
+        if(!$exeSource)
+        {
+            Write-Verbose "Falling back to x86 program files..." -Verbose
+            $exeSource = Get-ChildItem -path "${env:ProgramFiles(x86)}\$Executable" -Recurse -ErrorAction SilentlyContinue | Select-Object -First 1 -ExpandProperty FullName
+        }
+
+        if(!$exeSource)
+        {
+            Write-Verbose "Falling back to chocolatey..." -Verbose
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/dockerInstall.psm1]: Please explain order of lookup in comments.

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/dockerInstall.psm1 [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            $exeSource = Get-ChildItem -path "$env:ProgramData\chocolatey\$Executable" -Recurse -ErrorAction SilentlyContinue | Select-Object -First 1 -ExpandProperty FullName
+        }
+
+        if(!$exeSource)
+        {
+            Write-Verbose "Falling back to the root of the drive..." -Verbose
+            $exeSource = Get-ChildItem -path "/$Executable" -Recurse -ErrorAction SilentlyContinue | Select-Object -First 1 -ExpandProperty FullName
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/dockerInstall.psm1]: Please add comment for brute force

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1 [hunk 5]
```diff
@@ -0,0 +1,24 @@
+Import-Module "$PSScriptRoot\dockerInstall.psm1"
+
+function Install-WixZip
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1]: Please add a comment why we switched to zip instead on MSI, due to it requiring older version of dotnet

File: tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1 [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    param($zipPath)
+
+    $targetRoot = "${env:ProgramFiles(x86)}\WiX Toolset xcopy"
+    $binPath = Join-Path -Path $targetRoot -ChildPath 'bin'
+    if(-not (Test-Path $targetRoot))
+    {
+        $null = New-Item -Path $targetRoot -ItemType Directory
+    }
+    Write-Verbose "Expanding $zipPath to $binPath ..." -Verbose
+    Expand-Archive -Path $zipPath -DestinationPath $binPath
```
Comment (adityapatwardhan) [inline tools/releaseBuild/Images/microsoft_powershell_windowsservercore/wix.psm1]: -Force creates destination directory

File: tools/releaseBuild/generatePackgeSigning.ps1 [hunk 7]
```diff
@@ -42,11 +42,11 @@ function New-FileElement
     if(Test-Path -Path $file)
     {
         $name = Split-Path -Leaf -Path $File
-        $null = $fileElement = $XmlDoc.CreateElement("file")
+        $fileElement = $XmlDoc.CreateElement("file")
```
Comment (adityapatwardhan) [inline tools/releaseBuild/generatePackgeSigning.ps1]: Please exclude this change from this PR.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: tools/releaseBuild/generatePackgeSigning.ps1 (hunk 7 above)
Comment (TravisEz13) [reply tools/releaseBuild/generatePackgeSigning.ps1]: No, sorry this is a change that is needed.  Saw it in reverse when I talked to you.

**Comment by TravisEz13** [comment]
Verified release build

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**